src/Raven.Server/Documents/Handlers/Processors/Streaming/AbstractStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Raven.Client;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Exceptions.Documents;
+using Raven.Server.Documents.Queries;
+using Raven.Server.NotificationCenter;
+using Raven.Server.ServerWide;
+using Raven.Server.TrafficWatch;
+using Sparrow.Json;
+using HttpMethod = System.Net.Http.HttpMethod;
+
+namespace Raven.Server.Documents.Handlers.Processors.Streaming
+{
+    internal abstract class AbstractStreamingHandlerProcessorForGetStreamQuery<TRequestHandler, TOperationContext> : AbstractDatabaseHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+        where TOperationContext : JsonOperationContext
+    {
+        protected readonly HttpMethod _method;
+
+        protected AbstractStreamingHandlerProcessorForGetStreamQuery([NotNull] TRequestHandler requestHandler, HttpMethod method) : base(requestHandler)
+        {
+            _method = method;
```

Comment:
Can we assert here that `method` is Get or Post?

Reply:
done